apps/hash-api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -455,19 +454,23 @@ export const removeEntityEditorResolver: ResolverFn<
 const parseGqlAuthorizationViewerInput = ({
   kind,
   viewer,
-}: AuthorizationViewerInput): AccountId | AccountGroupId | "public" => {
+}: AuthorizationViewerInput) => {
   if (kind === AuthorizationSubjectKind.Public) {
-    return "public" as const;
+    return { namespace: "public" } as const;
   } else if (kind === AuthorizationSubjectKind.Account) {
     if (!viewer) {
       throw new UserInputError("Viewer Account ID must be specified");
     }
-    return viewer;
+    return { namespace: "account", id: viewer as AccountId } as const;
   } else {
     if (!viewer) {
       throw new UserInputError("Viewer Account Group ID must be specified");
     }
-    return viewer;
+    return {
+      namespace: "accountGroup",
+      id: viewer as AccountGroupId,
+      set: "member",
```

Comment:
We agreed to remove `set` as it will always be `member` if it's an account group, and nothing if it isn't, so it isn't adding information (and naming is hard)

We can reintroduce if we get more possible `set` values (e.g. `admin`)